### ISSUE TITLE
Please exports PaginatedSearchResult

### ISSUE THREAD

**Issue by frankmungnodev** [meilisearch/meilisearch-dart#242]
**Description**
I want to use meilisearch with pagination but the response was only searchable, that do not contain like totalPage field.

**Basic example**
 I have to `import 'package:meilisearch/src/paginated_search_result.dart';` which show warnings.
  ```
if (searchResult is PaginatedSearchResult) {
      int? nextPage = (searchResult.totalPages ?? 1) > page ? page++ : null;
      return nextPage;
    }
```

**Comment by brunoocasali** [comment]
Hi @frankmungnodev, thanks for raising this with me!

I have a fix for it, as you suggested, but I would like to be more confident that this doesn't happen again. Would you happen to have any tips that I can apply to my testing suite?

Another issue is regarding the other possible non-accessible things because, for example, `Result` is not accessible out of the scope of the library. Still, I don't know if it should be. :/

**Comment by frankmungnodev** [comment]
Hi @brunoocasali thank you for your quick response.  I will open a new issue if I find any other issues. I quite happy with meilisearch right now.

**Comment by brunoocasali** [comment]
I just released v0.8.0. I hope it is working for you now! And I'm happy to see you're happy with us! Thanks!
